### PR TITLE
Add FXIOS-8273 [v125] WebEngine: telemetry hooks

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8273)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18362)

## :bulb: Description

Early API hooks for providing telemetry to WebEngine clients. (WIP)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed, I updated documentation / comments for complex code and public methods
- [x] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by mattreaganmozilla** [comment]
@lmarceau This is a WIP but let me know if this is following along the lines of what you had in mind for the WebEngine APIs.

File: BrowserKit/Sources/WebEngine/EngineSession.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@ import Foundation
 /// Protocol representing a single engine session. In browsers usually a session corresponds to a tab.
 public protocol EngineSession {
     var delegate: EngineSessionDelegate? { get set }
+    var telemetryProxy: EngineTelemetryProxy? { get set }
```
Comment (lmarceau) [inline BrowserKit/Sources/WebEngine/EngineSession.swift:10]: Nice name! 🙌

**Review by lmarceau — COMMENTED** [review]
This is great! LGTM 🙌

**Comment by mattreaganmozilla** [comment]
@lmarceau I started looking into unit tests for this, though I'm not actually sure how that should work in practice yet. The telemetry hooks are all WebKit specific (currently) and depend on callbacks from the WKWebView. 🤔

**Comment by lmarceau** [comment]
@mattreaganmozilla after https://mozilla-hub.atlassian.net/browse/FXIOS-8477 is done maybe we can revisit?

**Comment by mattreaganmozilla** [comment]
@lmarceau Sounds good 👍

**Review by thatswinnie — APPROVED** [review]

**Review by lmarceau — APPROVED** [review]

**Pull request merged**